### DISCUSSION TITLE
Tap-hold repeat

### DISCUSSION THREAD

**Discussion by elisiano** [manna-harbour/miryoku#73] [Q&A] [answered]
I always re-implemented the Miryoku layout by myself in my fork, but since it became very configurable out of the box, I wanted to try and use the "stock" implementation.

I am used to have tap-hold repeat on my layouts (I disable auto shift).

I tried to achieve the same with some settings in `rules_custom.mk` and `config_custom.h` but I was unable to.

Is there an easy way to do so?

I tried to search but the only relevant thing I could find in the doc is:
> Tap-hold auto-repeat is disabled to permit faster tap-hold switching on thumbs, but thumb tap keys are mirrored onto some layers for use with auto-repeat. On other layers thumb keys are redefined with important functions for that layer.

edit: to be clear: I'm trying to get the home mods and the thumb cluster to behave like that, regular keys already work

**Comment by manna-harbour** [comment, marked as answer]
To enable tap hold auto repeat you can comment out this line: https://github.com/manna-harbour/miryoku_qmk/blob/b8cc3b3696d71090d06e350a333d61bf9fdff513/users/manna-harbour_miryoku/config.h#L16.  Later you'll be able to do it from `custom_config.h` or similar, see #47 which will also cover this kind of customisation.

It's disabled just because of the issue you quoted, but there is now a per key version of the option, so you could enable it and just disable it per key where needed if it causes issues.

In the stock configuration, for auto repeat on thumbs you can use the non layer tap versions on the other layers, such as backspace on Nav.  Also for backspace and delete, combining with control is generally preferable.  For auto repeat on alphas for vi nav you can use the cursor keys on Nav instead.

**Comment by elisiano** [comment]
I think it boiled down to adding this into `custom_config.h`:
```
#undef TAPPING_FORCE_HOLD
```

edit: I did not refresh the browser window, I did not notice your answer 😅. I will look into thumbs repeat on other layers, seems interesting and if I can get used to that I can revert this setting.

**Reply by manna-harbour** [reply]
The intention of `custom_config.h` is to support customisation without merge conflicts, but it's a work in progress and the filename or other implementation details might change (so it hasn't been documented yet).  So at the moment you can use it and watch out for changes, or just make the edit directly as mentioned above (and watch out for merge conflicts).

**Comment by elisiano** [comment]
so yeah, I've noticed that the only thumbs repeat keys I need are:
* backspace, which is defined in the navigation layer as you pointed out
* space, which is present in the function layer
* occasionally the delete key, which I could not find, but I think I can survive for now.

So I'm super stoked that I can use the default implementation, thanks a lot for your work.

Question (and let me know if this needs to be a separate discussion), what's the best way of keeping the branch up to date if I modify files under _your_ user?

I assume it's safe to keep rebasing as the `custom_*` files should never change in your branches, right?

**Reply by manna-harbour** [reply]
Nav has the same thumbs as Base, so enter, backspace, and delete.

For maintaining changes, see https://github.com/manna-harbour/miryoku/discussions/73#discussioncomment-2217641.  If you'd like to discuss it further you're welcome to make a new discussion and we can continue there.
